cmd/locate.go

```diff
@@ -37,17 +36,42 @@ func init() {
 }
 
 func Locate(
-	buildlistFile string,
 	locatedFile string,
 ) error {
-	var buildlist []gotool.Module
-	err := ReadJSONFile(buildlistFile, &buildlist)
+	tidyCmd := exec.Command("go", "mod", "tidy")
+	err := tidyCmd.Run()
+	if err != nil {
+		return fmt.Errorf("running go mod tidy: %w", err)
+	}
+
+	vendorCmd := exec.Command("go", "mod", "vendor")
+	err = vendorCmd.Run()
+	if err != nil {
+		return fmt.Errorf("running go mod vendor: %w", err)
+	}
+
+	modFileBytes, err := os.ReadFile("./go.mod")
+	if err != nil {
+		return fmt.Errorf("reading go mod file: %w", err)
+	}
+
+	modFile, err := modfile.Parse("go.mod", modFileBytes, nil)
 	if err != nil {
-		return fmt.Errorf("readinb buildlist file: %w", err)
+		return fmt.Errorf("parsing go mod file: %w", err)
+	}
+
+	// NOTE: This does not handle replaced or retracted modules
+	mods := make([]gotool.Module, 0)
+	for _, dep := range modFile.Require {
+		mods = append(mods, gotool.Module{
+			Path:     dep.Mod.Path,
+			Dir:      "./vendor/" + dep.Mod.Path,
```

Comment:
Prepend `./vendor/` so the third party license mining library knows where to find the source